src/main/java/org/candlepin/subscriptions/resource/ResourceUtils.java

```diff
@@ -149,19 +153,21 @@ static Pageable getPageable(Integer offset, Integer limit, Sort sort) {
      * @param usage string form of usage
      * @return Usage enum
      */
-    public static Usage sanitizeUsage(String usage) {
-        if (usage == null) {
+    public static Usage sanitizeUsage(UsageGenerated usage) {
+        if (Objects.isNull(usage)) {
             return Usage.ANY;
         }
-        Usage sanitizedUsage = Usage.fromString(usage);
-        // If the usage parameter is not one that we support, then throw an exception.
-        // If we don't, the query would default to UNSPECIFIED, which would be confusing.
-        if (StringUtils.hasLength(usage) && sanitizedUsage == Usage.UNSPECIFIED) {
+        Usage sanitizedUsage = Usage.fromString(usage.toString());
+
+        if (Objects.equals(sanitizedUsage, Usage.UNSPECIFIED) && StringUtils.hasLength(usage.toString())) {
```

Comment:
Couple questions here

1. Is there anything downstream that is relying on a BadRequestException being thrown to handle the flow of execution?
2. Can I delete this block (as well as the sanitizeServiceLevel equivalent further down)?

If there is logic reliant on this exception being thrown, this PR will need further work.  Since the Resource.java classes are expecting a query param to be an enum now, the requests get rejected upstream, and we wouldn't have a situation where an invalid/undefined value got this far.  Which is where my second question comes from...can I delete it then.

Reply:
1. I don't believe anything relies on catching that exception.  Here I think we're relying on the exception bubbling up to RESTEasy which handles it.
2. I think you can delete this, yes, since the sanitization is handled via type conversion to an enum.